test/measurement_calibration/generate_data.py

```diff
@@ -0,0 +1,226 @@
+# -*- coding: utf-8 -*-
+
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2020.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+"""
+Generate data for measurement calibration fitter tests
+"""
+
+import os
+import json
+from test.utils import convert_ndarray_to_list_in_data
+import qiskit
+from qiskit.ignis.mitigation.measurement import (CompleteMeasFitter, TensoredMeasFitter,
+                                                 complete_meas_cal, tensored_meas_cal,
+                                                 MeasurementFilter)
+from qiskit.providers.aer.noise import NoiseModel
+from qiskit.providers.aer.noise.errors.standard_errors import pauli_error
+
+# fixed seed for simulations
+SEED = 42
+
+
+def meas_calib_circ_creation():
+    """
+    create measurement calibration circuit and a GHZ state circuit for the tests
+
+    Returns:
+        QuantumCircuit: the measurement calibration circuit
+        list: the mitigation pattern
+        QuantumCircuit: ghz circuit with 5 qubits (3 are used)
+
+    """
+    meas_calibs, state_labels = complete_meas_cal(qubit_list=[1, 2, 3], qr=5)
+
+    # Choose 3 qubits
+    q_1 = 1
+    q_2 = 2
+    q_3 = 3
+    ghz = qiskit.QuantumCircuit(5, 3)
+    ghz.h(q_1)
+    ghz.cx(q_1, q_2)
+    ghz.cx(q_1, q_3)
+    ghz.measure(q_1, 0)
+    ghz.measure(q_2, 1)
+    ghz.measure(q_3, 2)
+    ghz.measure(q_3, 2)
+    return meas_calibs, state_labels, ghz
+
+
+def tensored_calib_circ_creation():
+    """
+    create tensored measurement calibration circuit and a GHZ state circuit for the tests
+
+    Returns:
+        QuantumCircuit: the tensored measurement calibration circuit
+        list: the mitigation pattern
+        QuantumCircuit: ghz circuit with 5 qubits (3 are used)
+
+    """
+    mit_pattern = [[2], [4, 1]]
+    qr = qiskit.QuantumRegister(5)
+    # Generate the calibration circuits
+    meas_calibs, mit_pattern = tensored_meas_cal(mit_pattern, qr=qr)
+
+    cr = qiskit.ClassicalRegister(3)
+    ghz_circ = qiskit.QuantumCircuit(qr, cr)
+    ghz_circ.h(qr[2])
+    ghz_circ.cx(qr[2], qr[4])
+    ghz_circ.cx(qr[2], qr[1])
+    ghz_circ.measure(qr[2], cr[0])
+    ghz_circ.measure(qr[4], cr[1])
+    ghz_circ.measure(qr[1], cr[2])
+    return meas_calibs, mit_pattern, ghz_circ
+
+
+def meas_calibration_circ_execution(shots: int, seed: int):
+    """
+    create measurement calibration circuits and simulates them with noise
```

Comment:
Either "create ... simulate" or "creates ... simulates". The first option is more preferable.